### ISSUE TITLE
todo.txt compatability

### ISSUE THREAD

**Issue by sjz** [tanepiper/sublime-todomanager#2]
Just installed this, and already finding it super useful.

Any chance this system can be made a bit more compatible with: https://github.com/ginatrapani/todo.txt-cli/wiki/The-Todo.txt-Format?

I think the only real deal-breaker is this:
`If a task starts with an x (case-sensitive lowercase) followed directly by a space, it is complete. Period.`
Your system seems to use a single asterix with no space.

Nice to have's might include:
- auto date inclusion for start and completion (which I see you have listed in future releases)
- converting addon type stuff such as line number and function name to the `key:value` format specified (not super necessary)
- auto inclusion of project name based on some regex option (useful if you have some folder structure like /a/b/c/projectname/src_stuff)

I am a bit new to both systems so there may be other ways to achieve these.
I am also happy to help implement any of these if you're happy for me to do so.

**Comment by tanepiper** [comment]
Yes, I'm looking to improve this over the next month now I've completed my freelance work, I can finally dedicate some itme to this.

I'm going to start by re-writing the code to use a Lexer rather than just Regex loops, as I did base this off Todo.txt format but as of yet don't support dates as an example (although it's on the todo)

I see your offering to help so if you have any ideas around this I'd appreciate any thoughts, and of course any code changes will be attributed :)

**Comment by tanepiper** [comment]
As per the addon rules in https://github.com/ginatrapani/todo.txt-cli/wiki/The-Todo.txt-Format

Change line number and function name to this format, add support for date

**Comment by sjz** [comment]
Hey-just thought: is the line number system that feasible?
Any deletions or additions of lines would need to shift the number accordingly. If the line of the comment is deleted, does it too get deleted, or does it just shift upwards?
Also: class names of methods would be cool :)
